test/augmentation/test_random_generator.py

```diff
@@ -71,6 +73,189 @@ def test_same_on_batch(self, seed, expected, device, dtype):
+                torch.as_tensor(
+                    saturation if saturation is not None else torch.tensor([0., 0.]), device=device, dtype=dtype),
+                torch.as_tensor(hue if hue is not None else torch.tensor([0., 0.]), device=device, dtype=dtype),
+            )(torch.Size([8]))
+
+    def test_random_gen(self, device, dtype):
+        # TODO(jian): crashes with pytorch 1.10, cuda and fp64
+        if torch_version_geq(1, 10) and "cuda" in str(device):
+            pytest.skip("AssertionError: Tensor-likes are not close!")
+        torch.manual_seed(42)
+        batch_size = 8
+        jitter_params = ColorJiggleGenerator(
+            brightness=torch.tensor([0.8, 1.2], device=device, dtype=dtype),
+            contrast=torch.tensor([0.7, 1.3], device=device, dtype=dtype),
+            saturation=torch.tensor([0.6, 1.4], device=device, dtype=dtype),
+            hue=torch.tensor([-0.1, 0.1], device=device, dtype=dtype),
+        )(torch.Size([batch_size]))
+
+        expected_jitter_params = {
+            'brightness_factor': torch.tensor(
+                [1.1529, 1.1660, 0.9531, 1.1837, 0.9562, 1.0404, 0.9026, 1.1175], device=device, dtype=dtype
+            ),
+            'contrast_factor': torch.tensor(
+                [1.2645, 0.7799, 1.2608, 1.0561, 1.2216, 1.0406, 1.1447, 0.9576], device=device, dtype=dtype
+            ),
+            'hue_factor': torch.tensor(
+                [0.0771, 0.0148, -0.0467, 0.0255, -0.0461, -0.0117, -0.0406, 0.0663], device=device, dtype=dtype
+            ),
+            'saturation_factor': torch.tensor(
+                [0.6843, 0.8156, 0.8871, 0.7595, 1.0378, 0.6049, 1.3612, 0.6602], device=device, dtype=dtype
+            ),
+            'order': torch.tensor([3, 2, 0, 1], device=device, dtype=dtype),
+        }
+
+        assert set(list(jitter_params.keys())) == {
+            'brightness_factor',
+            'contrast_factor',
+            'hue_factor',
+            'saturation_factor',
+            'order',
+        }, "Redundant keys found apart from \
+                'brightness_factor', 'contrast_factor', 'hue_factor', 'saturation_factor', 'order'"
+
+        assert_close(
+            jitter_params['brightness_factor'], expected_jitter_params['brightness_factor'], rtol=1e-4, atol=1e-4
+        )
+        assert_close(jitter_params['contrast_factor'], expected_jitter_params['contrast_factor'], rtol=1e-4, atol=1e-4)
+        assert_close(jitter_params['hue_factor'], expected_jitter_params['hue_factor'], rtol=1e-4, atol=1e-4)
+        assert_close(
+            jitter_params['saturation_factor'], expected_jitter_params['saturation_factor'], rtol=1e-4, atol=1e-4
+        )
+        assert_close(jitter_params['order'].to(dtype), expected_jitter_params['order'], rtol=1e-4, atol=1e-4)
+
+    def test_random_gen_accumulative_additive_additive(self, device, dtype):
+        # TODO(jian): crashes with pytorch 1.10, cuda and fp64
```

Comment:
Check if `test_random_gen_accumulative_additive_additive` still crashes in pytorch 1.10.1

Reply:
(I leave it as a comment for later)